### DISCUSSION TITLE
Useage with Kubernetes (.env fallback)

### DISCUSSION THREAD

**Discussion by shayneoneill** [br3ndonland/fastenv#24] [General]
Ok so theres bit of a gap here I cant quite work out.

The problem with .env files is they violate the 12 factor design principle off getting configuration from the environment, as they essentially hardcode the values.

Thats usually solved by ommitting the .env in deployment so that things like azure or kubernetes (and probably AWS's equivelents) can then supply those values in via the environment.

The problem is, fastenv's behavior when removing .env is to crash.

Am I missing something important here?

**Comment by br3ndonland** [comment]
> The problem is, fastenv's behavior when removing .env is to crash.
> 
> Am I missing something important here?

As shown in the README and the [docs on environment variables](https://fastenv.bws.bio/environment), fastenv works perfectly fine without a `.env` file. Can you share some more details about this "crash" behavior you're describing? By "fallback," do you mean that, if you can't download the `.env` file, you would want to read environment variables that have already been set?
